core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -200,6 +207,21 @@ public void onNext(HttpObject o) {
                     newHeaders.remove(HttpHeaderNames.CONTENT_LENGTH);
                 }
 
+                if (!newHeaders.contains(HttpHeaderNames.SERVER) &&
+                    config.defaultServerNameResponseHeader()) {
+                    final String verInfo = String.format("Armeria/%s (%s)",
+                                                         Version.identify().get("armeria").artifactVersion(),
+                                                         System.getProperty("os.name"));
+                    newHeaders.add(HttpHeaderNames.SERVER, verInfo);
+                }
+
+                if (!newHeaders.contains(HttpHeaderNames.DATE) &&
+                    config.defaultServerDateResponseHeader()) {
```

Comment:
Ditto: Could you check the `config` first, which is relatively cheaper than `headers.contains()`?

Reply:
Fixed it.